domain/model/state/controllerstate.go

```diff
@@ -1802,3 +1803,47 @@ AND    ot.type = $dbPermission.object_type
 	}
 	return nil
 }
+
+// GetAllActivatedModelsUUIDQuery returns a SQL statement that will return all activated models uuids.
+func (st *State) GetAllActivatedModelsUUIDQuery() string {
+	return "SELECT uuid from model WHERE activated = true"
+}
+
+// GetActivatedModelUUIDs returns the subset of model uuids from the supplied list that are activated.
+// If no models associated with the uuids are activated then an empty slice is returned.
+func (st *State) GetActivatedModelUUIDs(ctx context.Context, uuids []string) ([]coremodel.UUID, error) {
```

Comment:
If the len(uuids) == 0, are we expecting it to return nil or all activated model UUIDs?

Reply:
I think we should just return empty early.
```
if len(uuids) == 0 {
    return nil, nil
}
```